### DISCUSSION TITLE
D1 Mini v3 and v4 interchangeability

### DISCUSSION THREAD

**Discussion by RichardWilkinson** [universam1/iSpindel#577] [Q&A] [answered]
Hi, I think I have a faulty WEMOS D1 Mini board. Was working but now wont enter into config mode. I've tried reflashing the firmware via USB but no joy. Rather than spending money on a complete new iSpindel kit, I'd just like to remove the existing board and replace.
I'm finding it difficult to find what I believe are genuine D1 v3 boards. Does anyone know I can just replace the v3 board with a v4 board and flash the firmware with no other modifications? I know that the v4 board has a few differences (I2C socket and USB-C instead of micro USB).
Thanks

**Comment by mp-se** [comment, marked as answer]
I've used ESP8266 from the manufacturer AZDelivery that worked fine. I bought them from amazon in germany. I have also used esp8266 from this brand https://www.makershop.de/ (they have a storefront on ebay). I have bought over 30+ from these and had only one that was faulty (which they replaced).

**Reply by RichardWilkinson** [reply]
Thanks. AZDelivery could be an option as they are available on UK Amazon. The price for one v3 board however is about double the price of a v4 board from the Wemos Official Store on AliExpress, so if I can use a v4 board I'd rather do that.

**Reply by ErikdBr** [reply]
For little over 7 GBP including shipping to the UK you'll have an original Lolin D1 mini V3.1.0 from AZDelivery's own site and you'll have it probably a lot quicker than from Aliexpress:
https://www.makershop.de/plattformen/d1-mini/wemos-d1-mini-2/

![image](https://user-images.githubusercontent.com/48140503/159047388-eabd49d2-d2cf-43a6-8834-5853c55c7a21.png)

**Reply by RichardWilkinson** [reply]
Thanks. I'm in no rush as I have another working iSpindel. Does anyone know if the v4 DQ mini board would work? For the low price, I might just buy one and see.

**Reply by hobipivo** [reply]
Yes, a "Lolin D1 mini v4" board will work, but you have no way of disconnecting the power from the USB 5V to the 5V pin on the Mini-D1 board!
When assembled, the USB port must not be operated at the same time as the built-in battery!

**Reply by RichardWilkinson** [reply]
Thanks. Yes, I'm aware that the battery must be removed from the iSpindel when you have it connected via the D1 mini USB socket. I always make sure to do that.
Out of interest, why is this important? Can it damage the DQ mini board or will it just not flash firmware?

**Reply by mp-se** [reply]
> Thanks. Yes, I'm aware that the battery must be removed from the iSpindel when you have it connected via the D1 mini USB socket. I always make sure to do that.
> 
> Out of interest, why is this important? Can it damage the DQ mini board or will it just not flash firmware?

You will charge the battery with 5v thus damage and overheat it which is not a good thing.

**Reply by RichardWilkinson** [reply]
> 

Thanks. And the output from the TP4056 charger board will be less than that? I'm assuming so. I just followed kit instructions for the Cherry Phillip PCB.

**Reply by ErikdBr** [reply]
You could in fact burn your house down
https://github.com/universam1/iSpindel/issues/349

**Reply by mp-se** [reply]
> > 
> 
> 
> 
> Thanks. And the output from the TP4056 charger board will be less than that? I'm assuming so. I just followed kit instructions for the Cherry Phillip PCB.

Yes, if i remember correctly the battery it fully charged at 4.2 any more and it can take damage.

**Reply by thegreatgunbantoad** [reply]
I have never bothered to remove the lithium cell and had it attached for hours with no damage or even warming of it.  Since there is a diode in the way surely you wouldn't get 5V anyway you'd get a drop of ~0,7 V across the diode so the battery will see 4.3 maybe 4.4.  I expect the electronics hiding inside a decent modern protected lithium battery can handle that.  I do have some spare fake cells, maybe one day I'll over voltage them and see when they die.  (Sounds like an outside job under a plant pot)

[Edit:  Just put a voltmeter across without a battery in, 4.7 V.  More than I expected for sure]

**Reply by hobipivo** [reply]
Lilo Mini D1 Board v4 has no diode. USB-C supports up to 3A.
The USB-C 5V connects directly to an uncharged 3.2V battery.
The question is what breaks first? Track, module, battery ...

**Reply by thegreatgunbantoad** [reply]
Ahh,  sorry didn't catch that.  Well I doubt the Wemos is going to request the higher charging current (I believe authentication is needed for anything over the standard lowest current demand), so it won't be your track, I don't see why the module would be at risk, a protected battery should be fine though as the circuitry is there to stop the thing blowing up.  I would expect that the battery just might not charge at a higher voltage as the protection circuitry (Assuming it's a protected battery) may just cut the voltage to the lithium cell.  I would hope that the design of the battery's internal protection circuits are such that major over voltage would blow those circuits before it blows the battery, otherwise that's pretty rubbish protection.
